DDDEastAnglia.Tests/Helpers/LoginMethods/LoginMethodIconProviderTests.cs

```diff
@@ -0,0 +1,35 @@
+﻿using DDDEastAnglia.Helpers.AppSettings;
+using DDDEastAnglia.Helpers.LoginMethods;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace DDDEastAnglia.Tests.Helpers.LoginMethods
+{
+    [TestFixture]
+    public sealed class LoginMethodIconProviderTests
+    {
+        [Test]
+        public void GetIcon_ReturnsTheConfiguredIconForOAuthProvider()
+        {
+            var appSettingsProvider = Substitute.For<IAppSettingsProvider>();
+            appSettingsProvider.GetSetting(LoginMethodIconProvider.AppSettingsKeyPrefix + "test").Returns("test-icon");
```

Comment:
The flip side of that last remark is that we need to test the `AppSettingsProvider` is *called* correctly too - the appropriate method is called, with the expected arguments.